### ISSUE TITLE
Add capability to provide list of bad frames for 1- and 2-time correlation

### ISSUE THREAD

**Issue by ericdill** [scikit-beam/scikit-beam#394]
This was requested by @afluerasu on 2015-01-13. CHX needs the ability to remove entire frames from the correlator by providing a list of bad frames to functions.  @yugangzhang has code to do this and we should work with him to get his functionality integrated into scikit-beam.  It looks like #393 is providing this functionality for xray speckle visibility spectroscopy.  I wonder why it was closed?

@sameera2004

**Comment by sameera2004** [comment]
@ericdill, yes yesterday I discussed this with Andrei and Yugang. Yes, Yugang has a code for this, the issue was it needs for 1-time, 2-time and speckle visibility(XSVS). The code he has is taking a long time as an example if you done one time with 5min now it take more than 30min to do it. I was discussing this issue with @tacaswell in this morning and he asked me to use another way. Yugang has done this separately for 1time, 2 time , xsvs @tacaswell want to have one function to take care of this. Yugang gave me the correlation code and he updated the XSVS.

But this PR opening / closing - @yugangzhang opened it and he closed it by himself I am not sure why?

**Comment by yugangzhang** [comment]
It's not exactly correct  for what Sameera said for speed. We actually did not use the same data set. But, anyway I put all the codes on our xf11id machine. The path can be found in the beginning of one of our pipelines. That function dealing with bad frames  is our priority.

**Comment by ericdill** [comment]
Ok great. Thanks for the info regarding data sets to use.  I remember you have code that handles this problem of bad frames.  Could you share that code?  Or perhaps put in a pull request against scikit-beam with the changes that you need?

**Comment by yugangzhang** [comment]
@ericdill here is the codes

https://github.com/yugangzhang/CHX_CODE_DEVELOP

The one deal with bad frames is in Time_Correlation_Functions.py
The other two are for speckle analysis and new develops at chx.

**Comment by sameera2004** [comment]
@yugangzhang Thanks!, I started to work on it, using the new method @tacaswell told me, but I will follow the book keeping type method you used to remove the bad images. 

 As I remember last time we waited more than half an hour to  get the results, but we never waited that long for one time correlation data! , that's why I wanted to make it faster. Anyway I will send you the updated code hoping it will not take that long, I think it will take care of both correlation and xsvs for bad_farmes.

**Comment by yugangzhang** [comment]
That will great! Thanks!

**Comment by ambarb** [comment]
Are we still going to try to do this?  The Nexus-file type for data results will include a list of dropped frames for data provenance.  We might as well proceed with that here.

Is there also any information about @tacaswell method?
